src/ecs/rendering/systems.rs

```diff
@@ -58,10 +58,12 @@ impl<'a> System<'a> for RenderSystem {
             self.scene.add_light(light.0.clone());
         }
 
-        self.scene.add_camera(camera.clone());
+        let mut cam = camera.clone();
```

Comment:
Why the clone? You should be able to dereference with `*cam`.

Reply:
Ah, I see you copied the previous code. Your code makes it so only the version going into the renderer have the updated values, which is not what you want I think.

```
camera.update();
self.scene.add_camera(camera.clone());
```